src/helpers.ts

```diff
@@ -73,6 +81,112 @@ export function createInterestRate(
   return interestRate;
 }
 
+export function createIndexesUpdated(
+  blockNumber: BigInt,
+  timestamp: BigInt,
+  market: Market,
+  __MATHS__: IMaths
+): (_IndexesAndRatesHistory | null)[] {
+  const id: string = `${market.id.toHex()}-${blockNumber.toString()}`;
+  let indexesUpdated = _IndexesAndRatesHistory.load(id);
+
+  // This is an edge case when we're on the first block on the market has been created.
+  // We update the initial IndexesAndRatesHistory with the first data, and don't set the previous one.
+  if (indexesUpdated && !market._previousIndexesAndRatesHistory)
+    return [null, createOrSetInitialEmptyIndexesAndRatesHistory(blockNumber, timestamp, market)];
+
+  const lastInvariant = _IndexesAndRatesHistory.load(
+    indexesUpdated ? market._previousIndexesAndRatesHistory! : market._lastIndexesAndRatesHistory
+  );
+  if (!indexesUpdated) indexesUpdated = new _IndexesAndRatesHistory(id);
+  if (!lastInvariant) log.critical("No last invariant", []);
```

Comment:
```suggestion
  if (!lastInvariant) {
    log.critical("No last invariant", []);
    return []
  }
```

For type issues

Reply:
[Useless, `log.critical` terminates the subgraph](https://thegraph.com/docs/en/developing/assemblyscript-api/#logging-api)

Reply:
yes I  know,  but with the return keywork, you don't have to force type definition with  `!`, which is less errorprone